src/main/java/org/codeforamerica/messaging/models/Template.java

```diff
@@ -23,10 +24,21 @@ public class Template {
     @Column(unique=true)
     @ToString.Include
     String name;
-    @OneToMany(mappedBy = "template", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
-    List<TemplateVariant> templateVariants;
+    @ToString.Include
+    @Singular
+    @OneToMany(mappedBy = "template", fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true)
+    List<TemplateVariant> templateVariants = new LinkedList<>();
     @CreationTimestamp
     private OffsetDateTime creationTimestamp;
     @UpdateTimestamp
     private OffsetDateTime updateTimestamp;
+
+    public TemplateVariant addTemplateVariant(TemplateVariant.TemplateVariantBuilder templateVariantBuilder) {
```

Comment:
From https://vladmihalcea.com/jpa-hibernate-synchronize-bidirectional-entity-associations/, the recommended practice would be:
```
public TemplateVariant addTemplateVariant(TemplateVariant templateVariant) {
    templateVariants.add(templateVariant);
    templateVariant.setTemplate(this);
}
public TemplateVariant removeTemplateVariant(TemplateVariant templateVariant) {
    templateVariants.remove(templateVariant);
    templateVariant.setTemplate(null);
}

```

Reply:
What I am trying to avoid is that the `TemplateVariant` would have a `null` `Template` before being added to and after being removed from the `Template`.

Reply:
That shouldn't cause a problem because it would be a transient event. The `TemplateVariant` without a `Template` could never be persisted.